org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullAnnotationTests18.java

```diff
@@ -541,4 +541,28 @@ public void testBug576329() {
 		runner.expectedOutputString = "0";
 		runner.runConformTest();
 	}
+
+	@org.junit.Test
+	public void testNamedInstanceOfIsNonNull() {
+		Runner runner = getDefaultRunner();
+		runner.testFiles = new String[] {
+				"X.java",
+				"import org.eclipse.jdt.annotation.*;\n" +
+				"public class X {\n" +
+				"	public static void consumeNonNull(@NonNull String s) {\n" +
+				"		System.out.println(\"nonnull\");\n" +
+				"	}" +
```

Comment:
in case we observe a compile error (which I did with an experimental variant of this test), missing \n makes it hard to match line numbers.